### PR TITLE
Fix running `cstest` in CI

### PR DESCRIPTION
**Pull request by peace-maker**
All failures in the CITest workflow were ignored due to using `shell: 'script -q -e -c "bash {0}"'` 😱 
`cstest` is linked dynamically and couldn't find capstone because it wasn't getting installed.

The MC tests didn't parse the options in the comment lines leading to wrongly formatted output.

There appear to be some ARM related regressions that went unnoticed before.

### CODE REVIEW

**Comment by Rot127** [comment]
Yes, we were aware of it :( But we ignored it so far, because of time constraints and the testing in CS is very flawed in general. I think the `cstest_report.py` script not even fails when Capstone segfaults. It just reports 0 errors.

For he general discussion regarding testing see: https://github.com/capstone-engine/capstone/issues/1984.

The code currently in `next` is not untested. It was way more deeply tested with our Rizin tests.

Also ARM tests are only partly regressions (`cin`, thanks for that!). The `pop` ones are correct. Those were a bug in LLVM.

File: .github/workflows/CITest.yml [hunk 1]
```diff
@@ -81,37 +80,32 @@ jobs:
 
     - name: make
       if: startsWith(matrix.config.build-system, 'make')
-      shell: 'script -q -e -c "bash {0}"'
       run: |
-        ./make.sh;
-        make check;
+        ./make.sh
+        make check
+        sudo make install
```
Comment (Rot127) [inline .github/workflows/CITest.yml:86]: ~~If it makes not too much trouble please use cmake here. The `Makefile` build is deprecated.~~ Ouh, misread what workflow this is :D Ignore it

**Review by Rot127 — APPROVED** [review]

**Comment by Rot127** [comment]
@peace-maker Could you please cherry-pick the two commits from https://github.com/Rot127/capstone/tree/fix-arm-tests into this PR.

**Comment by Rot127** [comment]
Same as above for the detail tests. I no longer looked at them once I started to test these instructions in Rizin. Will fix them later or tomorrow.

**Comment by Rot127** [comment]
@peace-maker Sorry. Took me a while. You can cherry pick the 5 commits from https://github.com/Rot127/capstone/tree/arm-test

**Comment by peace-maker** [comment]
Maybe it'd be easier to merge and for you to fix the problems in a different branch? I don't feel competent in the ARM code to fix the issues myself and only wanted to get `cstest` running.

Now that we started this way I'll still continue cherry-picking if you want to too.

**Comment by Rot127** [comment]
Yes. I'll look at it again tomorrow. Of cause I forgot update the tests of the other issue files -.-
@kabeor You can merge if you want. Wouldn't make too much of a difference anyways.

**Comment by Rot127** [comment]
So I do not get any errors anymore after this https://github.com/peace-maker/capstone/pull/1
But who knows...

**Review by Rot127 — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by kabeor** [comment]
Looks wonderful. Thanks!

**Pull request merged**